xdsl/backend/riscv/lowering/convert_memref_to_riscv.py

```diff
@@ -176,11 +176,11 @@ def match_and_rewrite(self, op: memref.StoreOp, rewriter: PatternRewriter):
 
         rewriter.insert_op_before_matched_op(ops)
         match value.type:
-            case riscv.IntRegisterType():
+            case riscv.Registers.UNALLOCATED_INT:
```

Comment:
This one looks suspect, this was just being used for an isinstance check right?

Reply:
The tests passed because the registers were never allocated, which might actually be safe to assume here, but I'll remove this change from this PR